control/control_grpc_server.go

```diff
@@ -94,14 +95,14 @@ func (pc *ControlGRPCServer) CollectMetrics(ctx context.Context, r *rpc.CollectM
 func (pc *ControlGRPCServer) ValidateDeps(ctx context.Context, r *rpc.ValidateDepsRequest) (*rpc.ValidateDepsReply, error) {
 	metrics := common.ToRequestedMetrics(r.Metrics)
 	plugins := common.ToSubPlugins(r.Plugins)
-	configTree := common.ToConfigDataTree(r.CfgTree)
+	configTree := cdata.NewTree()
 	serrors := pc.control.ValidateDeps(metrics, plugins, configTree)
 	return &rpc.ValidateDepsReply{Errors: common.NewErrors(serrors)}, nil
 }
 
 func (pc *ControlGRPCServer) SubscribeDeps(ctx context.Context, r *rpc.SubscribeDepsRequest) (*rpc.SubscribeDepsReply, error) {
 	plugins := common.MsgToCorePlugins(r.Plugins)
-	configTree := common.ToConfigDataTree(r.CfgTree)
+	configTree := cdata.NewTree()
```

Comment:
Maybe leave a comment here about why it's ok to pass in empty tree (that the configTree is only used on collection and remote collection is not currently supported).